src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -1084,54 +1072,49 @@ mongoc_client_set_ssl_opts (mongoc_client_t *client,
 #endif
 
 
-/*
- *--------------------------------------------------------------------------
- *
- * mongoc_client_new_from_uri --
- *
- *       Create a new mongoc_client_t for a mongoc_uri_t.
- *
- * Returns:
- *       A newly allocated mongoc_client_t.
- *
- * Side effects:
- *       None.
- *
- *--------------------------------------------------------------------------
- */
-
 mongoc_client_t *
 mongoc_client_new_from_uri (const mongoc_uri_t *uri)
+{
+   mongoc_client_t *client;
+   bson_error_t error = {0};
+
+   if (!(client = mongoc_client_new_from_uri_with_error (uri, &error)) &&
+       0 != strcmp (error.message, "")) {
```

Comment:
```suggestion
   if (!(client = mongoc_client_new_from_uri_with_error (uri, &error))) {
```
`mongoc_client_new_from_uri_with_error ()` should return `NULL` on error, implying `error.code != 0`, so the `strcmp()` seems unnecessary.